### PR TITLE
Restructure targets and code

### PR DESCRIPTION
**Pull request by onmyway133**
Hi, long time no Puma. I've been thinking about this. Instead of the lengthy Options, let's use Set with convenient protocol extensions instead.

So the API looks like this

```swift
run {
    SetVersionNumber {
        $0.versionNumberForAllTargets("1.1")
    }
    
    SetBuildNumber {
        $0.buildNumberForAllTargets("2")
    }
    
    Build {
        $0.default(project: "TestApp", scheme: "TestApp")
        $0.buildsForTesting(enabled: true)
    }
    
    Test {
        $0.default(project: "TestApp", scheme: "TestApp")
        $0.testsWithoutBuilding(enabled: true)
        $0.destination(Destination(
            platform: Destination.Platform.iOSSimulator,
            name: Destination.Name.iPhoneXr,
            os: Destination.OS.os12_2
        ))
    }
}
```

First, Restructure our Package.swift into multiple modules Puma, PumaCore, PumaiOS and PumaAndroid

Second, add a macOS command line project to quickly test drive Puma, you will love this @besarism 

Third, introduce Deps as dependencies, and Console instead of Log

### CODE REVIEW

File: Example/TestPuma/TestPuma/main.swift [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+//  Created by khoa on 30/11/2019.
+//  Copyright © 2019 Khoa Pham. All rights reserved.
+//
+
+import Foundation
+import Puma
+import PumaiOS
+
+func testDrive() {
+//    let automaticSigning = AutomaticSigning(teamId: "T78DK947F2")
```
Comment (gabrielPeart) [inline Example/TestPuma/TestPuma/main.swift]: Can get rid of the commented out code

File: Sources/iOS/Tasks/Plist/SetBuildNumber.swift [hunk 2]
```diff
@@ -0,0 +1,31 @@
+//
+//  SetBuildNumber.swift
+//  Puma
+//
+//  Created by khoa on 15/04/2019.
+//  Copyright © 2019 onmyway133. All rights reserved.
```
Comment (gabrielPeart) [inline Sources/iOS/Tasks/Plist/SetBuildNumber.swift]: The copyright headers aren’t harmonious. Could be a good idea to keep the same consistent.

Comment (onmyway133) [reply Sources/iOS/Tasks/Plist/SetBuildNumber.swift]: Oh nice catch, I have no idea why there are such discrepancies 🙄

File: Example/TestPuma/TestPuma/main.swift (hunk 1 above)
Comment (onmyway133) [reply Example/TestPuma/TestPuma/main.swift]: Will do

File: Sources/iOS/Tasks/Plist/SetBuildNumber.swift (hunk 2 above)
Comment (Sajjon) [reply Sources/iOS/Tasks/Plist/SetBuildNumber.swift]: By adding an IDETemplateMacros.plist inside a xcshareddata folder inside project you can enforce same header for all new files. Like this:
https://github.com/radixdlt/radixdlt-swift/blob/develop/RadixSDK.xcodeproj/xcshareddata/IDETemplateMacros.plist

And here is a small bash script to change all the current files: https://stackoverflow.com/a/54336900/1311272

Comment (onmyway133) [reply Sources/iOS/Tasks/Plist/SetBuildNumber.swift]: @Sajjon Hi, thanks for the tip

**Pull request merged**